nikolai_chernyshov/main.cpp

```diff
@@ -105,34 +250,119 @@ class Graph {
 
 class GraphGenerator {
  public:
+  struct Params {
+    explicit Params(const int _depth = 0, const int _new_vertices_num = 0)
+        : depth(_depth), new_vertices_num(_new_vertices_num) {}
+
+    const int depth = 0;
+    const int new_vertices_num = 0;
+  };
+
+  explicit GraphGenerator(const Params& params = Params()) : params_(params) {}
+
+  Graph generate() const {
+    Graph graph;
+
+    add_gray_edges(graph);
+    add_green_edges(graph);
+    add_yellow_edges(graph);
+    add_red_edges(graph);
+
+    return graph;
+  }
+
+  void add_gray_edges(Graph& graph) const {
+    double proba_step = 1.0 / (double)params_.depth;
+    VertexId current_vertex_num = 0, added_vertex_num = 1;
+
+    graph.add_vertex();
+    for (int depth = 0; depth < params_.depth; depth++) {
+      for (int vertex_num = 0;
+           vertex_num < graph.get_vertices_in_depth(depth).size();
+           vertex_num++) {
```

Comment:
Ну что ты творишь. Вместо реальных данных. притягиваешь за уши какие-то другие
```suggestion
      const auto vertex_ids= graph.get_vertex_ids_in_depth(depth);
      for (const auto& vertex : vertex_ids) {
```